pusher/pusher_client.py

```diff
@@ -107,8 +107,23 @@ def trigger_batch(self, batch=[], already_encoded=False):
         """
         if not already_encoded:
             for event in batch:
-                event['data'] = data_to_string(event['data'],
-                self._json_encoder)
+
+                if not isinstance(event['channel'], six.string_types):
+                    raise TypeError("Channel name is expected to be a Python2 str or unicode or a Python 3 str")
```

Comment:
This should already be validated by `validate_channel()`

Reply:
Yes, it is. 👍